test/common/common.go

```diff
@@ -220,6 +220,20 @@ func MapContains(actual Object, expected Object) {
 	}
 }
 
+func RemoveResourceByCriterion(testCtx *TestContext, criterion query.Criterion, resourceType types.ObjectType) error {
+	if err := testCtx.SMRepository.InTransaction(context.TODO(), func(ctx context.Context, storage storage.Repository) error {
```

Comment:
Do you need to open the tx explicitly? won't it be created automatically in Delete?

Reply:
Sorry, looking in the IDE i would like to rephrase - why not use testCtx.SMRepository.Delete?

Reply:
yes this won't work as there will be an operation missing in the context